Winhttp flaky test (#5356)
Azure/azure-sdk-for-cpp: sdk/core/azure-core/CHANGELOG.md

### Bugs Fixed

> - Fixed and issue where WinHttp transport was not closing correctly in case of request timeout.

Reviewer: ```suggestion
- [[5244](https://github.com/Azure/azure-sdk-for-cpp/issues/5244)] Fixed an issue where WinHttp transport was not closing correctly in case of request timeout.
```

Reviewer: Formatting nit around the changelog.

> - [[5244](https://github.com/Azure/azure-sdk-for-cpp/issues/5244)] Fixed an issue where WinHttp transport was not closing correctly in case of request timeout.

Reviewer 2: ```suggestion
- [[5244]](https://github.com/Azure/azure-sdk-for-cpp/issues/5244) WinHTTP transport not closing correctly in case of a request timeout.
```
